examples/layer_shell.rs

```diff
@@ -0,0 +1,255 @@
+                    next_render_event_handle.set(Some(RenderEvent::Configure { width, height }));
+                }
+                (_, _) => {}
+            }
+        });
+
+        // Commit so that the server will send a configure event
+        surface.commit();
+
+        Self {
+            surface,
+            layer_surface,
+            next_render_event,
+            pools,
+            dimensions: (0, 0),
+        }
+    }
+
+    /// Handles any events that have occurred since the last call, redrawing if needed.
+    /// Returns true if the surface should be dropped.
+    fn handle_events(&mut self) -> bool {
+        match self.next_render_event.take() {
+            Some(RenderEvent::Closed) => true,
+            Some(RenderEvent::Configure { width, height }) => {
+                self.dimensions = (width, height);
+                self.draw();
+                false
+            }
+            None => false,
+        }
+    }
+
+    fn draw(&mut self) {
+        let pool = self.pools.pool().unwrap();
+
+        let stride = 4 * self.dimensions.0 as i32;
+        let width = self.dimensions.0 as i32;
+        let height = self.dimensions.1 as i32;
+
+        // First make sure the pool is the right size
+        pool.resize((stride * height) as usize).unwrap();
+
+        // Create a new buffer from the pool
+        let buffer = pool.buffer(0, width, height, stride, wl_shm::Format::Argb8888);
+
+        // Write the color to all bytes of the pool
+        pool.seek(SeekFrom::Start(0)).unwrap();
+        {
+            let mut writer = BufWriter::new(&mut *pool);
+            for _ in 0..(width * height) {
+                writer.write_u32::<NativeEndian>(0xff00ff00).unwrap();
+            }
+            writer.flush().unwrap();
+        }
+
+        // Attach the buffer to the surface and mark the entire surface as damaged
+        self.surface.attach(Some(&buffer), 0, 0);
+        self.surface
+            .damage_buffer(0, 0, width as i32, height as i32);
+
+        // Finally, commit the surface
+        self.surface.commit();
+    }
+}
+
+impl Drop for Surface {
+    fn drop(&mut self) {
+        self.layer_surface.destroy();
+        self.surface.destroy();
+    }
+}
+
+fn main() {
+    let (env, display, queue) = {
+        let display = Display::connect_to_env().unwrap();
+        let mut queue = display.create_event_queue();
+        let env = Environment::init(
+            &Proxy::clone(&display).attach(queue.token()),
+            Env {
+                compositor: SimpleGlobal::new(),
+                layer_shell: SimpleGlobal::new(),
+                shm: ShmHandler::new(),
+                outputs: OutputHandler::new(),
+            },
+        );
+        let ret = queue.sync_roundtrip(&mut (), |_, _, _| unreachable!());
+        ret.and_then(|_| queue.sync_roundtrip(&mut (), |_, _, _| unreachable!()))
+            .expect("Error during initial setup");
+
+        (env, display, queue)
+    };
```

Comment:
Similarly, I think this can be done using `init_default_environment!`:

```rust
let (env, display, queue) = init_default_environment!(Env,
    fields=[
        layer_shell: SimpleGlobal::new()
    ]
).expect("Unable to connect to the wayland compositor");
```